### PR TITLE
Completed the todo. fix #7

### LINKED ISSUE #7: todo elements for CFP
in current cfp there are 4 todo elements:

- [ ] talk duration
- [ ] all videos must be submitted to us within mid-January. TODO Discuss time
- [ ] Deadlines TODO discuss these
- [ ] Video recording submission is mid-January (TODO specify time)

I open this issue for discussion.

My opinions on the above:

- _(talk duration)_ looking at our past talks I see durations of 15', 20', 30'. The [Zig programming language CFP of last year](https://ziglang.org/news/fosdem-2021/) had as default 45'. I think our durations could range from 15'-45' including discussions. Once we have the durations we can build a schedule and I do not see any particular issue about a specific time slot to fill (half day should be 4 hours which could be 8 30' talks). As far as I understand we do not have to fill the 4 hours and if we have more materials we can ask for some time the day after
- _(within mid-January. TODO Discuss time)_ the 5th (day of event) is a Saturday. We could pick a specific deadline Jan 15 which is also a Saturday and 3 weeks before the conference. We will likely be not to strict about the deadline but it is good to have a set day and we will need I would say at least 2 weeks to review content and prepare the schedule
- _(Deadlines TODO discuss these)_ all deadlines reported seems fine (they are the official ones)
- _(TODO specify time)_ this is same todo as above

### PR DESCRIPTION
**Pull request by pietroppeter**
editing CFP taking into account discussion in https://github.com/pietroppeter/fosdem2022/issues/7

### CODE REVIEW

**Comment by pietroppeter** [comment]
mmh actually I think the deadline for submitting the recorded talk is too early. since selected talks are announced on Dec 31st you would only have 2 weeks to prepare and record. should we make it **January 22nd**? cc @PMunch, @dom96

File: cfp.md [hunk 1]
```diff
@@ -47,15 +47,23 @@ talks](https://www.youtube.com/channel/UCDAYn_VFt0VisL5-1a5Dk7Q):
 ### Requirements
 To be able to submit a talk you _must_ fulfil these formal requirements:
 
-- TODO Talk duration?
+- you should target a **talk duration** of 15' or 30'. We are also open to durations up to 45' for talks with a bigger scope.
+  We should be able to accomodate for durations that do not exactly hit the target.
 - Be able to pre-record a talk in advance. In order for us to verify the content
-all videos must be submitted to us within mid-January. TODO Discuss time
+all videos must be submitted by January 15th.
```
Comment (pietroppeter) [inline cfp.md]:
```suggestion
all videos must be submitted by January 22nd.
```

File: cfp.md [hunk 2]
```diff
@@ -47,15 +47,23 @@ talks](https://www.youtube.com/channel/UCDAYn_VFt0VisL5-1a5Dk7Q):
...
 presentation and the ensuing Q&A.
 
-### Deadlines TODO discuss these
+### Deadlines
 - Submission of talk: Dec 23rd, 2021
 - Selected talks announced: Dec 31st 2021
 - Conference dates 5 & 6 February 
 - The Nim language devroom is on the 5th
-- Video recording submission is mid-January (TODO specify time)
+- Video recording submission: January 15th
```
Comment (pietroppeter) [inline cfp.md]:
```suggestion
- Video recording submission: January 22nd
```

**Review by pietroppeter — COMMENTED** [review]
changes 15 -> 22

**Comment by PMunch** [comment]
Looks good to me. Merge this and publish?

File: cfp.md [hunk 3]
```diff
@@ -47,15 +47,23 @@ talks](https://www.youtube.com/channel/UCDAYn_VFt0VisL5-1a5Dk7Q):
...
 - The Nim language devroom is on the 5th
-- Video recording submission is mid-January (TODO specify time)
+- Video recording submission: January 22nd
+
+All deadlines are intended by 23.59 UTC.
+
+### Final remarks
+
+- make sure to check out [FOSDEM Code of Conduct](https://fosdem.org/2022/practical/conduct/)
+- for any questions about this call for participation you can reach us in #fosdem channel on our Nim Discord (also available in Matrix as "Nim FOSDEM")
```
Comment (dom96) [inline cfp.md]: Can we add links to Discord and/or Matrix?

File: cfp.md [hunk 4]
```diff
@@ -47,15 +47,23 @@ talks](https://www.youtube.com/channel/UCDAYn_VFt0VisL5-1a5Dk7Q):
...
-### Deadlines TODO discuss these
+### Deadlines
 - Submission of talk: Dec 23rd, 2021
 - Selected talks announced: Dec 31st 2021
 - Conference dates 5 & 6 February 
 - The Nim language devroom is on the 5th
-- Video recording submission is mid-January (TODO specify time)
+- Video recording submission: January 22nd
+
+All deadlines are intended by 23.59 UTC.
```
Comment (dom96) [inline cfp.md]: "intended by" sounds odd to me

```suggestion
All deadline times are 23.59 UTC.
```

File: cfp.md [hunk 5]
```diff
@@ -47,15 +47,23 @@ talks](https://www.youtube.com/channel/UCDAYn_VFt0VisL5-1a5Dk7Q):
 ### Requirements
 To be able to submit a talk you _must_ fulfil these formal requirements:
 
-- TODO Talk duration?
+- you should target a **talk duration** of 15' or 30'. We are also open to durations up to 45' for talks with a bigger scope.
```
Comment (dom96) [inline cfp.md]:
```suggestion
- you should target a **talk duration** of 15 or 30 minutes. We are also open to durations up to 45 minutes for talks with a bigger scope, but we would encourage sticking to 15/30 minute durations.
```

File: cfp.md [hunk 6]
```diff
@@ -47,15 +47,23 @@ talks](https://www.youtube.com/channel/UCDAYn_VFt0VisL5-1a5Dk7Q):
...
+  We should be able to accomodate for durations that do not exactly hit the target.
```
Comment (dom96) [inline cfp.md]:
```suggestion
  We should be able to accommodate for durations that do not exactly hit the target.
```

**Review by dom96 — COMMENTED** [review]
A few nits from me.

**Comment by dom96** [comment]
Overall looks good, we're publishing to nim-lang.org though right?

File: cfp.md (hunk 3 above)
Comment (pietroppeter) [reply cfp.md]:
```suggestion
- for any questions about this call for participation you can reach us in [#fosdem channel](https://discord.com/channels/371759389889003530/909464670098833409) on our Nim Discord (also available in Matrix as "[Nim FOSDEM](https://matrix.to/#/#nim-fosdem:matrix.org)")
```

**Comment by pietroppeter** [comment]
nits most welcome. the tricky part was the links, can you try if you are able to use them?

yes, I would agree that we should publish to nim-lang. If we are satisfied after this PR is merged we should open a PR to nim website (the whole CfP could have probably been a PR to nim website from the start)

**Comment by PMunch** [comment]
Did some nitpicking myself. You can check how the document renders here: https://github.com/pietroppeter/fosdem2022/blob/pietroppeter-complete-todo/cfp.md.

And yes publish to nim-lang.org maybe with a cross-post and link to the forum (I think copy the text to the forum, with an additional link, just a link on the forum doesn't look good IMO). And of course links in the usual places.

**Comment by dom96** [comment]
I think on the forum we should just give a TL;DR and then link to the blog post :)

**Comment by PMunch** [comment]
Then we have to write a TL;DR :sweat_smile:

**Comment by pietroppeter** [comment]
Honestly the text being so short it is hard to do a TLDR (I do agree that just a link on forum is not great). We could skip sections about FOSDEM, Call for participation, final remarks (keeping intro, requirements, deadlines), mentioning full announcement on website.

**Comment by dom96** [comment]
How about something like:

> We're excited to announce that for FOSDEM 2022 there will be a Nim Programming Language developer room This means we need talks and lots of them. The talks will be pre-recorded so it's a good opportunity for anyone who may not be confident speaking in front of a live audience. For more details check out: `<blog post>`.

The reason I don't think we should copy/paste the article into the forum is that AFAIK Google then demotes it in its search results (plus it's also confusing about what is the canonical announcement).

Also I think the forum is a good opportunity to thank all of you for helping with the organisation of this :)

**Pull request merged**